win32ss/user/ntuser/callback.c

```diff
@@ -1272,4 +1272,45 @@ co_IntImmProcessKey(HWND hWnd, HKL hKL, UINT vKey, LPARAM lParam, DWORD dwHotKey
     return ret;
 }
 
+// Win: ClientImmLoadLayout
+BOOL APIENTRY co_ClientImmLoadLayout(HKL hKL, PIMEINFOEX pImeInfoEx)
+{
+    NTSTATUS Status;
+    IMMLOADLAYOUT_CALLBACK_ARGUMENTS Common = { hKL };
+    ULONG ResultLength = sizeof(IMMLOADLAYOUT_CALLBACK_OUTPUT);
+    PVOID ResultPointer = NULL;
+    PIMMLOADLAYOUT_CALLBACK_OUTPUT Output;
+
+    ERR("hKL: %p\n", hKL);
+
+    RtlZeroMemory(pImeInfoEx, sizeof(IMEINFOEX));
+
+    UserLeave();
+    Status = KeUserModeCallback(USER32_CALLBACK_IMMLOADLAYOUT, &Common, sizeof(Common),
+                                &ResultPointer, &ResultLength);
+    UserEnterExclusive();
+
+    if (!NT_SUCCESS(Status))
+    {
+        ERR("!NT_SUCCESS\n");
+        return FALSE;
+    }
+
+    if (ResultLength != sizeof(IMMLOADLAYOUT_CALLBACK_OUTPUT))
+    {
+        ERR("ResultLength\n");
+        return FALSE;
+    }
+
+    Output = ResultPointer;
+    if (!Output->ret)
+    {
+        ERR("!ret\n");
+        return FALSE;
+    }
+
+    RtlCopyMemory(pImeInfoEx, &Output->iiex, sizeof(IMEINFOEX));
```

Comment:
Wowowow.... In which address space does the data pointed by `Output` reside?

Reply:
???

Reply:
Used SEH2 and `ProbeForRead`.